neurodsp/aperiodic/autocorr.py

```diff
@@ -45,3 +46,151 @@ def compute_autocorr(sig, max_lag=1000, lag_step=1, demean=True):
     timepoints = np.arange(0, max_lag+1, lag_step)
 
     return timepoints, autocorrs
+
+
+def compute_decay_time(timepoints, autocorrs, fs, level=0):
+    """Compute autocorrelation decay time, from precomputed autocorrelation.
+
+    Parameters
+    ----------
+    timepoints : 1d array
+        Timepoints for the computed autocorrelations.
+    autocorrs : 1d array
+        Autocorrelation values.
+    fs : int
+        Sampling rate of the signal.
+    level : float
+        Autocorrelation decay threshold.
+
+    Returns
+    -------
+    result : float
+        Autocorrelation decay time.
+        If decay time value found, returns nan.
+
+    Notes
+    -----
+    The autocorrelation decay time is computed as the time delay for the autocorrelation
+    to drop to (or below) the decay time threshold.
+    """
+
+    val_checks = autocorrs <= level
+
+    if np.any(val_checks):
+        # Get the first value to cross the threshold, and convert to time value
+        result = timepoints[np.argmax(val_checks)] / fs
+    else:
+        result = np.nan
+
+    return result
+
+
+def fit_autocorr(timepoints, autocorrs, fs=None, fit_function='single_exp'):
+    """Fit autocorrelation function, returning timescale estimate.
+
+    Parameters
+    ----------
+    timepoints : 1d array
+        Timepoints for the computed autocorrelations.
+    autocorrs : 1d array
+        Autocorrelation values.
+    fs : int, optional
+        Sampling rate of the signal.
+        If provided, timepoints are converted to time values.
+    fit_func : {'single_exp', 'double_exp'}
+        Which fitting function to use to fit the autocorrelation results.
+
+    Returns
+    -------
+    popts
+        Fit parameters.
+    """
+
+    if fs:
+        timepoints = timepoints / fs
+
+    if fit_function == 'single_exp':
+        p_bounds =([0, 0, 0], [10, np.inf, np.inf])
+    elif fit_function == 'double_exp':
+        p_bounds =([0, 0, 0, 0, 0], [10, np.inf, np.inf, np.inf, np.inf])
+
+    popts, _ = curve_fit(AC_FIT_FUNCS[fit_function], timepoints, autocorrs, bounds=p_bounds)
+
+    return popts
+
+
+## AC FITTING
+
+def exp_decay_func(timepoints, tau, scale, offset):
+    """Exponential decay fit function.
+
+    Parameters
+    ----------
+    timepoints : 1d array
+        Time values.
+    tau : float
+        Timescale value.
+    scale : float
+        Scaling factor, which captures the start value of the function.
+    offset : float
+        Offset factor, which captures the end value of the function.
+
+    Returns
+    -------
+    ac_fit : 1d array
+        Results of fitting the function to the autocorrelation.
+    """
+
+    return scale * (np.exp(-timepoints / tau) + offset)
```

Comment:
Another way to write what you point out is that if `fs` is passed into the fit function, then the `timepoints` should be in time not in samples, and so recomputing the values consistently should look like (this is equivalent to multiplying the tau):
`corrs = exp_decay_func(lags / fs, tau, 1, 0)`

Reply:
Okay, the core of this issue is basically it's not a good idea to optionally take `fs` as an input in `fit_autocorr` and if so divide it out, because then it makes the values of the fit parameters different from the values of the inputs. I'll drop this, and then one can pass in timepoints as either samples or seconds, and it will work either way (and be consistent).